pdg/src/info.rs

```diff
@@ -67,18 +67,483 @@ fn set_flow_info(g: &mut Graph) {
+        //|      C2
+        //B2
+        let a = mk_addr_of_local(&mut g, 0_u32);
+        let b1 = mk_copy(&mut g, a);
+        let c11 = mk_field(&mut g, a, 1_u32);
+        let c1 = mk_copy(&mut g, c11);
+        let c2 = mk_store_addr(&mut g, c1);
+        let b2 = mk_store_addr(&mut g, b1);
+
+        let pdg = build_pdg(g);
+        assert!(!info(&pdg, a).unique);
+        assert!(!info(&pdg, b1).unique);
+        assert!(!info(&pdg, b2).unique);
+        assert!(!info(&pdg, c1).unique);
+        assert!(!info(&pdg, c2).unique);
+    }
+
+    #[test]
+    fn fields_different_levels() {
+        let mut g = Graph::default();
+        //let mut a = Point {x: 0, y:0}; // A
+        //let b = &mut a;                // B1
+        //let c = &mut b.y;              // C1
+        //let bb = &mut b.y;             // B2
+        //*b = 1;                        // B3
+        //*c = 2;                        // C
+        //
+        //A
+        //|-------
+        //|      |
+        //B1     |y
+        //|      C1
+        //|y
+        //B2
+        //
+        let a = mk_addr_of_local(&mut g, 0_u32);
+        let b1 = mk_copy(&mut g, a);
+        let c1 = mk_field(&mut g, a, 1_u32);
+        let b2 = mk_field(&mut g, b1, 1_u32);
+        let c2 = mk_store_addr(&mut g, c1);
+
+        let pdg = build_pdg(g);
+        assert!(!info(&pdg, a).unique);
+        assert!(!info(&pdg, b1).unique);
+        assert!(!info(&pdg, b2).unique);
+        assert!(!info(&pdg, c1).unique);
+        assert!(!info(&pdg, c2).unique);
+        assert!(!info(&pdg, b2).unique);
+    }
+
+    #[test]
+    fn lots_of_siblings() {
+        let mut g = Graph::default();
+        //let mut a = ColorPoint {x: 0, y: 0, z: Color{ r: 100, g: 100, b: 100}};   //A1
+        //let b = &mut a.x;                                                         //B1
+        //let c = &mut a.y;                                                         //C1
+        //a.z.r = 200;                                                              //A2
+        //*b = 4;                                                                   //B2
+        //*c = 2;                                                                   //C2
+        //let d = &mut a;                                                           //D1
+        //*d = ColorPoint {x: 0, y: 0, z: Color {r: 20, g:200, b: 20}};             //D2
+        //let e = &mut a.z;                                                         //E
+        //let f = &mut e.g;                                                         //F1
+        //let g = &mut e.g;                                                         //G
+        //*f = 3;                                                                   //F2
+        //a.z.r = 100;                                                              //A3
+
+        let a = mk_addr_of_local(&mut g, 0_u32);
+        let b1 = mk_field(&mut g, a, 0_u32);
+        let c1 = mk_field(&mut g, a, 1_u32);
```

Comment:
i think it would be helpful to have variables such as
```rust
let (x, y, z) = (0_u32, 1_u32, 2_u32);
```
to correspond to the comments more accurately